core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -144,6 +152,40 @@ class LogCleaner(val config: CleanerConfig,
     cleaners.foreach(_.shutdown())
```

Comment:
Maybe I'm missing it, but don't we need to clear `cleaners`? Perhaps there should be a test assertion somewhere.

Reply:
Oops, fixed and added test